### DISCUSSION TITLE
Error: Invalid Grant with /api/token on local build from source

### DISCUSSION THREAD

**Discussion by xingster** [fief-dev/fief#488] [Q&A]
Encountered a weird issue with a local build (with default env configurations) from mainline source build. Endpoint /api/token always returns "invalid grant" when trying to exchange auth code for a token. This error occurs when trying to access admin portal (e.g., "http://localhost:8000/admin")

I traced this to token.py::validate_grant_request() where the code_verifier errors out in this line: 
```
elif code_verifier is not None: 
        raise TokenRequestException(TokenError.get_invalid_grant()).
```
I added this to the top of validate_grant_request(): 
```
if code_verifier == "":
            logging.info("Code verifier is empty string")
            code_verifier = None
```
and it seems to be working now.

What was very strange was that this works on codespace and Docker images without any problems.

**Comment by frankie567** [comment]
Hey 👋 Did you pull the latest imageb/source code on your machine? This looks like the bug we fixed in #469 🙂

Try also to update the dependencies (`hatch env prune`, `hatch install`)
